owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt

```diff
@@ -0,0 +1,699 @@
+                }
+            } else {
+                accountEndAccountAvatar?.visibility = View.GONE
+                accountMiddleAccountAvatar?.visibility = View.GONE
+            }
+        }
+    }
+
+    /**
+     * re-populates the account list.
+     *
+     * @param accounts list of accounts
+     */
+    private fun repopulateAccountList(accounts: Array<Account>) {
+        // remove all accounts from list
+        nav_view.menu.removeGroup(R.id.drawer_menu_accounts)
+
+        // add all accounts to list
+        for (account in accounts) {
+            if (getAccount().name != account.name) {
+                val accountMenuItem: MenuItem =
+                    nav_view.menu.add(R.id.drawer_menu_accounts, Menu.NONE, MENU_ORDER_ACCOUNT, account.name)
+                accountMenuItem.loadAvatarForAccount(
+                    account = account,
+                    fetchIfNotCached = false,
+                    displayRadius = menuAccountAvatarRadiusDimension
+                )
+            }
+        }
+
+        // re-add add-account and manage-accounts
+        nav_view.menu.add(
+            R.id.drawer_menu_accounts, R.id.drawer_menu_account_add,
+            MENU_ORDER_ACCOUNT_FUNCTION,
+            resources.getString(R.string.prefs_add_account)
+        ).setIcon(R.drawable.ic_plus_grey)
+        nav_view.menu.add(
+            R.id.drawer_menu_accounts, R.id.drawer_menu_account_manage,
+            MENU_ORDER_ACCOUNT_FUNCTION,
+            resources.getString(R.string.drawer_manage_accounts)
+        ).setIcon(R.drawable.ic_group)
+
+        // adding sets menu group back to visible, so safety check and setting invisible
+        showMenu()
+    }
+
+    /**
+     * Updates the quota in the drawer
+     */
+    private fun updateQuota() {
+        Timber.d("Update Quota")
+        val account = AccountUtils.getCurrentOwnCloudAccount(this) ?: return
+        drawerViewModel.getStoredQuota(account.name)
+        drawerViewModel.userQuota.observe(this, Observer { event ->
+            when (event.peekContent()) {
+                is UIResult.Success -> {
+                    event.peekContent().getStoredData()?.let { userQuota ->
+                        when {
+                            userQuota.available < 0 -> { // Pending, unknown or unlimited free storage
+                                account_quota_bar?.visibility = View.VISIBLE
+                                account_quota_bar?.progress = 0
+                                account_quota_text?.text = String.format(
+                                    getString(R.string.drawer_unavailable_free_storage),
+                                    DisplayUtils.bytesToHumanReadable(userQuota.used, this)
+                                )
+                            }
+                            userQuota.available == 0L -> { // Quota 0, guest users
+                                account_quota_bar?.visibility = View.GONE
```

Comment:
You could use `account_quota_bar.isVisible = false` and same for the rest of places where you use View.Whatever.

Anyway, I think that `account_quota_bar` is not nullable and `account_quota_text` either so maybe you don't need ?

Reply:
Is nullable, unfortunately 😢